ff/src/fields/macros.rs

```diff
@@ -115,6 +116,117 @@ macro_rules! impl_Fp {
             pub const fn new(element: $BigIntegerType) -> Self {
                 Self(element, PhantomData)
             }
+
+            #[ark_ff_asm::unroll_for_loops]
+            const fn const_is_zero(&self) -> bool {
```

Comment:
Should all changes to field arithmetic now be duplicated in these macros?

Reply:
Hmm I don't anticipate these methods will change much, but if you're asking whether we should have a comment on the structs indicating that both copies should be updated if either one is, then yes. Longer term, the goal would be to deduplicate the code, and just use one impl in both const and non-const contexts.

Reply:
Cool! Can you add a comment to that affect for each one

Reply:
Will do

(maybe I'll even do a partial deduplication now)

Reply:
I did a partial deduplication, so that common logic is stored in a separate method that both the `const` and non-`const` versions invoke.